File: backend/app/schemas/customer.py
from pydantic import BaseModel


class CustomerBase(BaseModel):
    name: str
    mobile_number: str
    address: str
    aadhaar_number: str
    notes: str | None = None


class CustomerCreate(CustomerBase):
    pass


class CustomerUpdate(CustomerBase):
    pass


class CustomerOut(CustomerBase):
    id: int

    class Config:
        from_attributes = True

File: finance_app/lib/models/customer.dart
class Customer {
  final int id;
  final String name;
  final String mobileNumber;
  final String address;
  final String aadhaarNumber;
  final String? notes;

  Customer({required this.id, required this.name, required this.mobileNumber, required this.address, required this.aadhaarNumber, this.notes});

  factory Customer.fromJson(Map<String, dynamic> json) => Customer(
        id: json['id'],
        name: json['name'],
        mobileNumber: json['mobile_number'],
        address: json['address'],
        aadhaarNumber: json['aadhaar_number'],
        notes: json['notes'],
      );
}
